omnigibson/simulator.py

```diff
@@ -489,53 +594,36 @@ def import_scene(self, scene):
             assert self.is_stopped(), "Simulator must be stopped while importing a scene!"
             assert isinstance(scene, Scene), "import_scene can only be called with Scene"
 
-            # Clear the existing scene if any
-            self.clear()
-
-            # Initialize all global updatable object states
-            for state in self.object_state_types_requiring_update:
-                if issubclass(state, GlobalUpdateStateMixin):
-                    state.global_initialize()
+            # Check that the scene is not already imported
+            if scene in self._scenes:
```

Comment:
if we're going to do membership checks shall we make `self._scenes` either a set or dictionary instead of a list?

Reply:
We do need to do scene indexing at some places though: e.g. RigidContactAPI. But if you think it's worth it, we can definitely make this a dict. I don't think this membership check would be crazy slow though just because we won't have a huge number of scenes.

Reply:
This check can be replaced with something that simply checks if the scene has been loaded (e.g. it has a scene.loaded flag or something right?) so that scenes that were loaded into a previous simulator also cannot be reloaded. That way it'll be O(1) too.